src/core.js

```diff
@@ -18,17 +18,6 @@ extend( QUnit, {
 			}
 		}
 
-		// DEPRECATED: handles setup/teardown functions,
-		// beforeEach and afterEach should be used instead
-		if ( testEnvironment && testEnvironment.setup ) {
-			testEnvironment.beforeEach = testEnvironment.setup;
```

Comment:
This should throw an error as well, otherwise the `setup` and `teardown` properties are simply ignored, breaking tests in unexpected ways.